src/qml/Tracking.qml

```diff
@@ -0,0 +1,317 @@
+import QtQuick 2.11
+import QtQuick.Controls 2.4
+import QtQuick.Controls 1.4 as Controls
+import QtQuick.Dialogs 1.2
+import org.qgis 1.0
+import org.qfield 1.0
+import Theme 1.0
+import QtQuick.Layouts 1.3
+
+import '.'
+
+
+Item{
+    id: tracking
+
+    property VectorLayer trackingLayer //layerTree.data(layerTree.index(index,0), LayerTreeModel.VectorLayer )
+    property bool activated: false
+    property bool running: false
+    property var feature: undefined
+
+    signal trackingStarted
+    signal trackingStopped
+
+    RubberbandModel {
+        id: rubberbandModel
+        frozen: false
+        currentCoordinate: positionSource.projectedPosition
+        vectorLayer: trackingLayer
+        crs: mapCanvas.mapSettings.destinationCrs
+
+        onVertexCountChanged: {
+          if( ( geometryType === QgsWkbTypes.LineGeometry && vertexCount > 2 ) ||
+              ( geometryType === QgsWkbTypes.PolygonGeometry &&vertexCount > 3 ) )
+          {
+              featureModel.applyGeometry()
+
+              if( ( geometryType === QgsWkbTypes.LineGeometry && vertexCount == 3 ) ||
+                  ( geometryType === QgsWkbTypes.PolygonGeometry && vertexCount == 4 ) )
+              {
+                  featureModel.create()
+                  feature = featureModel.feature
+              }
+              else
+              {
+                  featureModel.save()
+              }
+          }
+        }
+    }
+
+    Rubberband {
+        id: rubberband
+        width: 4 * dp
+        color: Qt.rgba(Math.random(),Math.random(),Math.random(),0.6);
+
+        mapSettings: mapCanvas.mapSettings
+
+        model: rubberbandModel
+
+        anchors.fill: parent
+
+        visible: layerTree.data(layerTree.index(index,0), LayerTreeModel.Visible) //binding does not work and I don't know why
+    }
+
+    Track {
+        id: track
+
+        model: rubberbandModel
+    }
+
+    FeatureModel {
+        id: featureModel
+        currentLayer: trackingLayer
+        geometry: Geometry {
+          id: featureModelGeometry
+          rubberbandModel: rubberbandModel
+          vectorLayer: trackingLayer
+        }
+    }
+
+    function start( layer )
+    {
+        //layer is passed since I cannot bind trackingLayer with model.VectorLayer - don't know why
+        trackingLayer = layer
+        featureModel.resetAttributes()
+        embeddedFeatureForm.state = 'Add'
+        embeddedFeatureForm.active = true
```

Comment:
For another task (fill ring map tool), I would need to use a feature form to create a new feature too.
Do you think we could move this logic to somewhere it could be reused?